libs/server/Custom/CustomFunctions.cs

```diff
@@ -71,9 +72,9 @@ protected static unsafe void WriteBulkStringArray(ref MemoryResult<byte> output,
             fixed (byte* ptr = output.MemoryOwner.Memory.Span)
             {
                 var curr = ptr;
-                RespWriteUtils.WriteArrayLength(values.Length, ref curr, ptr + totalLen);
-                for (int i = 0; i < values.Length; i++)
-                    RespWriteUtils.WriteBulkString(values[i].Span, ref curr, ptr + totalLen);
+                _ = RespWriteUtils.WriteArrayLength(values.Length, ref curr, ptr + totalLen);
```

Comment:
I like that you added a note to remember that we are using pre-allocated memory and that `WriteArrayLength` should never fail. I would add that comment here as well, and maybe consume the output of the call to `RespWriteUtils.WriteArrayLength` and check in a debug assert (applies to the other locations as well, of course).